include/libtorrent/receive_buffer.hpp

```diff
@@ -155,7 +155,7 @@ struct TORRENT_EXTRA_EXPORT receive_buffer
 
 	// keep track of how much of the receive buffer we use, if we're not using
 	// enough of it we shrink it
-	sliding_average<std::int64_t, 20> m_watermark;
+	sliding_average<std::ptrdiff_t, 20> m_watermark;
```

Comment:
I wonder that the warning was here

Reply:
the mean value is used to specify a new size for a buffer, which means it should be a "buffer size" type, not `int64_t`. On second though though, I wonder if this increases the risk of overflow on 32 bit systems